publish-mqtt/src/main.rs

```diff
@@ -187,77 +191,150 @@ impl Sensor {
                 Self::PROPERTY_ID_TEMPERATURE,
                 format!("{:.2}", readings.temperature),
             )
-            .await?;
+            .await
+            .with_context(|| std::line!().to_string())?;
```

Comment:
- [ ] this was really useful for testing, but maybe we can use stable-eyre and set RUST_LIB_BACKTRACE=1 to get the same benefit for less typing?